var/spack/repos/builtin/packages/python-venv/package.py

```diff
@@ -93,6 +93,13 @@ def setup_dependent_run_environment(self, env, dependent_spec):
             path = os.path.join(dependent_spec.prefix, directory)
             if os.path.isdir(path):
                 env.prepend_path("PYTHONPATH", path)
+        windows = self.spec.satisfies("platform=windows")
+        dependent_bin_path = join_path(dependent_spec.prefix, "Scripts" if windows else "bin")
```

Comment:
Could you move the understanding of this into `PythonExtension` and condition this on `isinstance(PythonExtension)`? I admit that's a bit awkward, but I want `py-*` packages in general to be aware of the possibility that their binaries may live in `<prefix>/Scripts` on Windows.

Another option is to add a `python_bindir` attribute to the specs of packages that are `PythonExtension`s (in which case you can try to access that attribute and guard against `AttributeError`).

Reply:
I pushed a `try/except` approach as I thought that was more robust than a conditional on a base class also maybe non python packages want to implement a bindir method if they produce bindings.